lib/ui/views/homepage/homepage_view.dart

```diff
@@ -18,37 +20,73 @@ class HomepageView extends StackedView<HomepageViewModel> {
     HomepageViewModel viewModel,
     Widget? child,
   ) {
+    ThemeData theme = Theme.of(context);
+    AppTypography? typography = theme.extension<AppTypography>();
+    Palette? palette = theme.extension<Palette>();
     return Scaffold(
-      appBar: customAppBar(context, titleText: 'Home'),
+      // appBar: customAppBar(context, titleText: 'Home'),
+      appBar: AppBar(
+          backgroundColor: kcBackground,
+          elevation: 0,
+          leading: InkWell(
+              onTap: () {
+                //  TODO open drawer
+                Drawer();
+              },
+              child: Align(
+                alignment: Alignment.center,
+                child: Padding(
+                  padding: EdgeInsets.only(
+                    left: 16.w,
+                    top: 8.h,
+                  ),
+                  child: SvgPicture.asset(
+                    AppImages.burgerLog,
+                    width: 24.7.w,
+                    height: 24.h,
+                  ),
+                ),
+              )),
+          title: Text(
+            'Home',
+            style: typography?.titleBold16?.copyWith(color: palette?.gray11),
+          )),
       body: SafeArea(
         child: Column(
           children: [
             SizedBox(
-              height: 24.h,
+              height: 18.h,
             ),
             Expanded(
               child: SizedBox(
-                  // height: 400.h,
                   child: GridView.builder(
                 padding: EdgeInsets.symmetric(horizontal: sidePadding),
                 gridDelegate: SliverGridDelegateWithFixedCrossAxisCount(
                   crossAxisCount: 2,
-                  childAspectRatio: 171.w / 252.h,
+                  childAspectRatio: 0.67.h,
                   mainAxisSpacing: 20.h,
                   crossAxisSpacing: 16.w,
                   mainAxisExtent: 252.h,
                 ),
-                itemCount: viewModel.productItems().length,
+                itemCount: viewModel.productItems.length,
                 itemBuilder: (context, index) {
+            
                   return ProductItem(
-                    productModel: viewModel.productItems()[index],
+                    productModel: viewModel.productItems[index],
                   );
                 },
               )),
             ),
           ],
         ),
       ),
+      floatingActionButton: FloatingActionButton(
+        onPressed: () {
+//TODO route to add product
+        },
```

Comment:
create a function for this in the ViewModel and add it here... so it becomes `onPressed: viewModel.newFunction,` with newFuntion being the name of the function you created in the view model